pinot-core/src/main/java/com/linkedin/pinot/core/startree/operator/StarTreeFilterOperator.java

```diff
@@ -207,8 +205,9 @@ public BaseFilterBlock getNextBlock() {
     } else if (numChildFilterOperators == 1) {
       return childFilterOperators.get(0).nextBlock();
     } else {
-      FilterOperatorUtils.reOrderFilterOperators(childFilterOperators, _debugOptions);
-      return new AndOperator(childFilterOperators).nextBlock();
+      FilterOperatorUtils.reorderFilterOperators(childFilterOperators,
+          FilterOperatorUtils.enableScanReorderOptimization(_debugOptions));
```

Comment:
Reading this in the context of reorderFilterOperators definitely feels odd to me. Why not pass debugOptions and let reorderFilterOperator decide what it wants to look at?

Reply:
IMO, reorderFilterOperators method should be independent of debugOptions, and thus make future usage or testing easier.